Replicator/c4RemoteReplicator.hh

```diff
@@ -170,7 +170,7 @@ namespace c4Internal {
 
             // If this is a transient error, or if I'm continuous and the error might go away with
             // a change in network (i.e. network down, hostname unknown), then go offline.
-            bool transient = c4error_mayBeTransient(c4err);
+            bool transient = c4error_mayBeTransient(c4err) || (c4err.domain == WebSocketDomain && c4err.code == 4001);
```

Comment:
I have trouble with this because now we are basically forcing the consumer to rely on an implementation detail.  I think it would be better to document that 4001 is a special code and then handle it inside of `c4error_mayBeTransient`